### DISCUSSION TITLE
Creo2urdf issue to use the plugin

### DISCUSSION THREAD

**Discussion by Agrobotdev** [robotology/community#688] [Q&A] [answered]
I am trying to work with creo2urdf on my Creo 9.0.4:  

When I use the plugin directly on the example 2bars supplied I have this error : BAR misses the frame in the linkFrames section, CSYS will be used instea. For more clarty you can refer to the file attached error_1.png.  
![error_1](https://github.com/robotology/community/assets/160733967/f502857c-fe02-4dfb-b633-ed26cbee41e1)

When I modify the .yaml file to specify LinkFrame, I have this error : Element BAR is not present in the configuration file!
For more clarty you can refer to the file attached error_2.png.
![error_2](https://github.com/robotology/community/assets/160733967/8ec1df61-78d4-4d7f-a3b8-e0f4bf48162e)

 Is it a problem of Creo version ?

**Comment by Nicogene** [comment]
Hi @Agrobotdev!

> When I use the plugin directly on the example 2bars supplied I have this error : BAR misses the frame in the linkFrames section, CSYS will be used instea. For more clarty you can refer to the file attached error_1.png.

This is actually a warning, that for a simple mechanism as 2bars can be safely ignored, also because we didn't defined link frames for that assembly. This warning is preventing to generate the urdf?

cc @mfussi66

**Reply by Agrobotdev** [reply]
Hi Nicogene,

Thanks for your feedback!
Yes, nothing has been created and Creo closes itself.

**Reply by mfussi66** [reply]
Hi @Agrobotdev , could you post the `traceback.log` file that the window mentioned? That is most likely the cause of the closure of Creo.

**Reply by Agrobotdev** [reply]
Hi @mfussi66,  

yes of course, here is the traceback.log:
[traceback.log](https://github.com/robotology/community/files/14385688/traceback.log)

**Comment by mfussi66** [comment]
Thanks @Agrobotdev , I can see that the crash is caused by `EXCEPTION_NONCONTINUABLE_EXCEPTION`. Running the plugin on the same example in two different machines with Creo 9.0.4.0 did not crash the program, so we cannot really reproduce the bug at the moment.

The PTC forum has this [community discussion](https://community.ptc.com/t5/3D-Part-Assembly-Design/EXCEPTION-NONCONTINUABLE-EXCEPTION/td-p/765266) in which a user solved the same problem by wiping all Creo files and doing a clean reinstallation. 
Is it possible for you to do it?

**Reply by Agrobotdev** [reply]
Many thanks @mfussi66 ,  okay I will try that and let you know.

**Comment by Agrobotdev** [comment, marked as answer]
Hi @mfussi66 , 
I have tried but it's given me the result. I have contacted the IT service and I am waiting for their feedback. 
In the mean time I have installed Simscape Multibody link, but I don't know wher to run on windows:  simmechanics_to_urdf {SimMechanics XML filename} --yaml [yaml_configfile] --csv-joint csv_joints_configfile --output {xml|graph|none}

Thanks

**Reply by Nicogene** [reply]
> In the mean time I have installed Simscape Multibody link,

A curiosity, are you experiencing any issue running Simscape Multibody link,?

> In the mean time I have installed Simscape Multibody link, but I don't know wher to run on windows: simmechanics_to_urdf {SimMechanics XML filename} --yaml [yaml_configfile] --csv-joint csv_joints_configfile --output {xml|graph|none}

I always used it on linux but I think it can be also run via powershell, @traversaro can you confirm?

**Reply by Agrobotdev** [reply]
No, I have been able to add the plugin and did some test of generating XML file and it works. 

Okay I will try it on linux also.

**Reply by traversaro** [reply]
@Agrobotdev which version of MATLAB are you using? `simmechanics_to_urdf` is only compatible with quite old MATLAB releases, see the warning in the README: https://github.com/robotology/simmechanics-to-urdf?tab=readme-ov-file#simmechanics-to-urdf- .

**Reply by Agrobotdev** [reply]
I use Matlab 2023a. Okay thanks @traversaro

**Reply by mfussi66** [reply]
Thanks for the test @Agrobotdev . I wonder, do you have the C++ Object Toolkit (called OTK C++) installed? It is a requirement to be able to use the .dll .

**Reply by Agrobotdev** [reply]
Yes @mfussi66 , I guess since I found the otk_cpp folder on my disk.

**Reply by Agrobotdev** [reply]
What are the parameters necessary in the Yalm and csv file to create a urdf file with the plugin creo2urdf ?

**Reply by Nicogene** [reply]
The ones contained in this folder

- https://github.com/icub-tech-iit/creo2urdf/tree/master/examples/2bars

have already all the necessary parameters.
Which version of `creo2urdf` are you using? This is the latest:
- https://github.com/icub-tech-iit/creo2urdf/releases/tag/v0.4.5

**Reply by Agrobotdev** [reply]
Hi @Nicogene , okay I meant even with a completely new mechanism ? 
Alright, I think this is the one that I've installed, but I will reinstall this one and let you know. 
Best

**Reply by Nicogene** [reply]
> okay I meant even with a completely new mechanism ?

The parameters expressed here should be the only mandatory:
- https://github.com/icub-tech-iit/creo2urdf/blob/master/examples/2bars/2bars.yaml

But depending on what you need to export in your URDF you will need to specify also other parameters (e.g. sensors, export frames, rename links/joints)

**Reply by Nicogene** [reply]
Hi @Agrobotdev,

you may try with the dll of the latest release, it should work now:
- https://github.com/icub-tech-iit/creo2urdf/releases/tag/v0.4.6

See:
- https://github.com/icub-tech-iit/creo2urdf/issues/84#issuecomment-2007639276
